diff --git a/node_modules/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuReactHost.kt b/node_modules/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuReactHost.kt
index 9f17865..c881932 100644
--- a/node_modules/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuReactHost.kt
+++ b/node_modules/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuReactHost.kt
@@ -3,7 +3,6 @@ package expo.modules.devmenu
 import android.app.Application
 import android.content.Context
 import android.util.Log
-import com.facebook.react.JSEngineResolutionAlgorithm
 import com.facebook.react.ReactHost
 import com.facebook.react.ReactPackage
 import com.facebook.react.bridge.JSBundleLoader
@@ -14,7 +13,6 @@ import com.facebook.react.defaults.DefaultTurboModuleManagerDelegate
 import com.facebook.react.devsupport.DevMenuReactSettings
 import com.facebook.react.devsupport.DevServerHelper
 import com.facebook.react.fabric.ComponentFactory
-import com.facebook.react.runtime.JSCInstance
 import com.facebook.react.runtime.ReactHostImpl
 import com.facebook.react.runtime.hermes.HermesInstance
 import com.facebook.react.shell.MainReactPackage
@@ -36,12 +34,7 @@ object DevMenuReactHost {
     val jsBundleAssetPath = "EXDevMenuApp.android.js"
     val jsBundleLoader =
       JSBundleLoader.createAssetLoader(application, "assets://$jsBundleAssetPath", true)
-    val jsResolutionAlgorithm = createJSEngineResolutionAlgorithm(application)
-    val jsRuntimeFactory = if (jsResolutionAlgorithm == JSEngineResolutionAlgorithm.JSC) {
-      JSCInstance()
-    } else {
-      HermesInstance()
-    }
+    val jsRuntimeFactory = HermesInstance()
     val jsMainModuleName = "packages/expo-dev-menu/app/index"
     val defaultReactHostDelegate =
       DefaultReactHostDelegate(
@@ -94,14 +87,6 @@ object DevMenuReactHost {
     return packages
   }

-  private fun createJSEngineResolutionAlgorithm(application: Application): JSEngineResolutionAlgorithm {
-    SoLoader.init(application.applicationContext, /* native exopackage */ false)
-    if (SoLoader.getLibraryPath("libjsc.so") != null) {
-      return JSEngineResolutionAlgorithm.JSC
-    }
-    return JSEngineResolutionAlgorithm.HERMES
-  }
-
   /**
    * To use a different packager url, we need to replace internal RN objects.
    */
diff --git a/node_modules/expo-dev-menu/android/src/main/java/com/facebook/react/devsupport/DevMenuSettingsBase.kt b/node_modules/expo-dev-menu/android/src/main/java/com/facebook/react/devsupport/DevMenuSettingsBase.kt
index 4079443..ba65397 100644
--- a/node_modules/expo-dev-menu/android/src/main/java/com/facebook/react/devsupport/DevMenuSettingsBase.kt
+++ b/node_modules/expo-dev-menu/android/src/main/java/com/facebook/react/devsupport/DevMenuSettingsBase.kt
@@ -55,9 +55,6 @@ abstract class DevMenuSettingsBase(
       mPreferences.edit().putBoolean("remote_js_debug", remoteJSDebugEnabled).apply()
     }

-  override var isStartSamplingProfilerOnInit: Boolean =
-    mPreferences.getBoolean("start_sampling_profiler_on_init", false)
-
   override fun addMenuItem(title: String) {
   }

diff --git a/node_modules/expo-dev-menu/android/src/main/java/expo/modules/devmenu/react/DevMenuHostHelper.kt b/node_modules/expo-dev-menu/android/src/main/java/expo/modules/devmenu/react/DevMenuHostHelper.kt
index f8a960e..771c80f 100644
--- a/node_modules/expo-dev-menu/android/src/main/java/expo/modules/devmenu/react/DevMenuHostHelper.kt
+++ b/node_modules/expo-dev-menu/android/src/main/java/expo/modules/devmenu/react/DevMenuHostHelper.kt
@@ -3,15 +3,8 @@ package expo.modules.devmenu.react
 import android.app.Application
 import com.facebook.hermes.reactexecutor.HermesExecutorFactory
 import com.facebook.react.bridge.JavaScriptExecutorFactory
-import com.facebook.react.jscexecutor.JSCExecutorFactory
-import com.facebook.react.modules.systeminfo.AndroidInfoHelpers
-import com.facebook.soloader.SoLoader

 fun createNonDebuggableJavaScriptExecutorFactory(application: Application): JavaScriptExecutorFactory {
-  SoLoader.init(application.applicationContext, /* native exopackage */ false)
-  if (SoLoader.getLibraryPath("libjsc.so") != null) {
-    return JSCExecutorFactory(application.packageName, AndroidInfoHelpers.getFriendlyDeviceName())
-  }
   return HermesExecutorFactory().also {
     try {
       HermesExecutorFactory::class.java
