diff --git a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherReactHost.kt b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherReactHost.kt
index 914cf69..423e685 100644
--- a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherReactHost.kt
+++ b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/DevLauncherReactHost.kt
@@ -1,7 +1,6 @@
 package expo.modules.devlauncher.launcher
 
 import android.app.Application
-import com.facebook.react.JSEngineResolutionAlgorithm
 import com.facebook.react.ReactHost
 import com.facebook.react.ReactPackage
 import com.facebook.react.bridge.JSBundleLoader
@@ -10,7 +9,6 @@ import com.facebook.react.defaults.DefaultComponentsRegistry
 import com.facebook.react.defaults.DefaultReactHostDelegate
 import com.facebook.react.defaults.DefaultTurboModuleManagerDelegate
 import com.facebook.react.fabric.ComponentFactory
-import com.facebook.react.runtime.JSCInstance
 import com.facebook.react.runtime.ReactHostImpl
 import com.facebook.react.runtime.hermes.HermesInstance
 import com.facebook.react.shell.MainReactPackage
@@ -34,12 +32,7 @@ object DevLauncherReactHost {
     val jsBundleAssetPath = "expo_dev_launcher_android.bundle"
     val jsBundleLoader =
       JSBundleLoader.createAssetLoader(application, "assets://$jsBundleAssetPath", true)
-    val jsResolutionAlgorithm = createJSEngineResolutionAlgorithm(application)
-    val jsRuntimeFactory = if (jsResolutionAlgorithm == JSEngineResolutionAlgorithm.JSC) {
-      JSCInstance()
-    } else {
-      HermesInstance()
-    }
+    val jsRuntimeFactory = HermesInstance()
     val jsMainModuleName = "packages/expo-dev-launcher/bundle/index"
     val defaultReactHostDelegate =
       DefaultReactHostDelegate(
@@ -94,11 +87,4 @@ object DevLauncherReactHost {
       additionalPackages
   }
 
-  private fun createJSEngineResolutionAlgorithm(application: Application): JSEngineResolutionAlgorithm {
-    SoLoader.init(application.applicationContext, /* native exopackage */ false)
-    if (SoLoader.getLibraryPath("libjsc.so") != null) {
-      return JSEngineResolutionAlgorithm.JSC
-    }
-    return JSEngineResolutionAlgorithm.HERMES
-  }
 }
diff --git a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgeDevSupportManager.kt b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgeDevSupportManager.kt
index b128d36..4a9c7a3 100644
--- a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgeDevSupportManager.kt
+++ b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgeDevSupportManager.kt
@@ -50,7 +50,7 @@ class DevLauncherBridgeDevSupportManager(
   @get:JvmName("getjSBundleURLForRemoteDebugging")
   override val jSBundleURLForRemoteDebugging: String? = null
 
-  override fun showNewJavaError(message: String?, e: Throwable?) {
+  override fun showNewJavaError(message: String?, e: Throwable) {
     Log.e("DevLauncher", "$message", e)
     if (!DevLauncherController.wasInitialized()) {
       Log.e(
@@ -67,11 +67,11 @@ class DevLauncherBridgeDevSupportManager(
     }
 
     controller?.onAppLoadedWithError()
-    val cause: Throwable? = if (e is InvocationTargetException) e.cause ?: e else e
+    val cause: Throwable = if (e is InvocationTargetException) e.cause ?: e else e
     DevLauncherErrorActivity.showError(activity, DevLauncherAppError(message, cause))
   }
 
-  override fun getUniqueTag() = "DevLauncherApp-Bridge"
+  fun getUniqueTag() = "DevLauncherApp-Bridge"
 
   override fun startInspector() {
     // no-op for the default `startInspector` which would be implicitly called
diff --git a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgelessDevSupportManager.kt b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgelessDevSupportManager.kt
index 9a117c1..37c7beb 100644
--- a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgelessDevSupportManager.kt
+++ b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherBridgelessDevSupportManager.kt
@@ -49,7 +49,7 @@ class DevLauncherBridgelessDevSupportManager(
   @get:JvmName("getjSBundleURLForRemoteDebugging")
   override val jSBundleURLForRemoteDebugging: String? = null
 
-  override fun showNewJavaError(message: String?, e: Throwable?) {
+  override fun showNewJavaError(message: String?, e: Throwable) {
     Log.e("DevLauncher", "$message", e)
     if (!DevLauncherController.wasInitialized()) {
       Log.e("DevLauncher", "DevLauncher wasn't initialized. Couldn't intercept native error handling.")
@@ -66,7 +66,7 @@ class DevLauncherBridgelessDevSupportManager(
     DevLauncherErrorActivity.showError(activity, DevLauncherAppError(message, e))
   }
 
-  override fun getUniqueTag() = "DevLauncherApp-Bridgeless"
+  fun getUniqueTag() = "DevLauncherApp-Bridgeless"
 
   override fun startInspector() {
     // no-op for the default `startInspector` which would be implicitly called
diff --git a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherReactNativeHostHandler.kt b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherReactNativeHostHandler.kt
index ff4dc69..c940d27 100644
--- a/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherReactNativeHostHandler.kt
+++ b/node_modules/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/react/DevLauncherReactNativeHostHandler.kt
@@ -4,8 +4,6 @@ import android.content.Context
 import com.facebook.hermes.reactexecutor.HermesExecutorFactory
 import com.facebook.react.bridge.JavaScriptExecutorFactory
 import com.facebook.react.devsupport.DevSupportManagerFactory
-import com.facebook.react.jscexecutor.JSCExecutorFactory
-import com.facebook.react.modules.systeminfo.AndroidInfoHelpers
 import com.facebook.soloader.SoLoader
 import expo.modules.core.interfaces.ReactNativeHostHandler
 import expo.modules.devlauncher.DevLauncherController
@@ -35,9 +33,6 @@ class DevLauncherReactNativeHostHandler(context: Context) : ReactNativeHostHandl
       // return null here to use the default value.
       return null
     }
-    if (SoLoader.getLibraryPath("libjsc.so") != null) {
-      return JSCExecutorFactory(applicationContext.packageName, AndroidInfoHelpers.getFriendlyDeviceName())
-    }
     return HermesExecutorFactory()
   }
 
diff --git a/node_modules/expo-dev-launcher/android/src/main/java/com/facebook/react/devsupport/DevLauncherDevServerHelper.kt b/node_modules/expo-dev-launcher/android/src/main/java/com/facebook/react/devsupport/DevLauncherDevServerHelper.kt
index 52ad415..4009133 100644
--- a/node_modules/expo-dev-launcher/android/src/main/java/com/facebook/react/devsupport/DevLauncherDevServerHelper.kt
+++ b/node_modules/expo-dev-launcher/android/src/main/java/com/facebook/react/devsupport/DevLauncherDevServerHelper.kt
@@ -33,19 +33,19 @@ class DevLauncherDevServerHelper(
       .build()
   }
 
-  override fun getDevServerBundleURL(jsModulePath: String?): String {
+  override fun getDevServerBundleURL(jsModulePath: String): String {
     return controller?.manifest?.getBundleURL() ?: super.getDevServerBundleURL(jsModulePath)
   }
 
-  override fun getDevServerSplitBundleURL(jsModulePath: String?): String {
+  override fun getDevServerSplitBundleURL(jsModulePath: String): String {
     return controller?.manifest?.getBundleURL() ?: super.getDevServerSplitBundleURL(jsModulePath)
   }
 
-  override fun getSourceUrl(mainModuleName: String?): String {
+  override fun getSourceUrl(mainModuleName: String): String {
     return controller?.manifest?.getBundleURL() ?: super.getSourceUrl(mainModuleName)
   }
 
-  override fun getSourceMapUrl(mainModuleName: String?): String {
+  override fun getSourceMapUrl(mainModuleName: String): String {
     val defaultValue = super.getSourceMapUrl(mainModuleName)
     val bundleURL = controller?.manifest?.getBundleURL()
       ?: return defaultValue
diff --git a/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/react/activitydelegates/DevLauncherReactActivityNOPDelegate.kt b/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/react/activitydelegates/DevLauncherReactActivityNOPDelegate.kt
index 1fd1d2c..b8d2bc1 100644
--- a/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/react/activitydelegates/DevLauncherReactActivityNOPDelegate.kt
+++ b/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/react/activitydelegates/DevLauncherReactActivityNOPDelegate.kt
@@ -16,6 +16,6 @@ open class DevLauncherReactActivityNOPDelegate(activity: ReactActivity) :
   override fun onNewIntent(intent: Intent?): Boolean = true
   override fun onBackPressed(): Boolean = true
   override fun onWindowFocusChanged(hasFocus: Boolean) {}
-  override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>?, grantResults: IntArray?) {}
-  override fun onConfigurationChanged(newConfig: Configuration?) {}
+  override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {}
+  override fun onConfigurationChanged(newConfig: Configuration) {}
 }
diff --git a/node_modules/expo-dev-launcher/expo-dev-launcher-gradle-plugin/build.gradle.kts b/node_modules/expo-dev-launcher/expo-dev-launcher-gradle-plugin/build.gradle.kts
index fe48821..48cc956 100644
--- a/node_modules/expo-dev-launcher/expo-dev-launcher-gradle-plugin/build.gradle.kts
+++ b/node_modules/expo-dev-launcher/expo-dev-launcher-gradle-plugin/build.gradle.kts
@@ -1,7 +1,7 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-  kotlin("jvm") version "1.9.24"
+  kotlin("jvm") version "2.1.20"
   id("java-gradle-plugin")
 }
 
